src/geometry.rs

```diff
@@ -1,3 +1,115 @@
+//!
+//! ```text
+//!            (U, V)                                         3    
+//!              3                    3------> N            / ^     
+//!            / |                  / |                   /   |     
+//!  (U, V)  /   |                /   |                 2     |     
+//!        2     |              2------> N                \   |     
+//!        | \   |              | \   |                 2   \ |     
+//!        |   \ |              |   \ |                 | \   1     
+//!        |     1              |     1------> N        |   \       
+//!        |   /   (U, V)       |   /                   |     1     
+//!        | /                  | /                     |   /       
+//!        0                    0------> N              V /         
+//!     (U, V)                                          0           
+//!
+//! Fig 2: Quad with vertices [A, B, C, D].
+//! - (left) UVs.
+//! - (middle) Normals.
+//! - (right) Two triangles rendered using indices [0, 1, 2, 3, 2, 1].
+//! ```
+//!
+//! # Coordinate System
+//!
+//! Different domains (e.g., your game engine or 3D modeling application) use
+//! different coordinate systems. Generating correct vertex positions, normals,
+//! and UV coordinates requires knowing the specifics of the coordinate system.
+//!
+//! ```text
+//!       +Y                 +Y                 -Y      
+//!       | +Z               | -Z               | -Z    
+//! -X____|/____+X     -X____|/____+X     +X____|/____-X
+//!      /|                 /|                 /|       
+//!    -Z |               +Z |               +Z |       
+//!       -Y                 -Y                 +Y      
+//!
+//! Fig 3: Various coordinate systems.
+//! - (left) Left-handed coordinate system with Y up.
+//! - (middle) Right-handed coordinate system with Y up.
+//! - (right) Right-handed coordinate system with Y down.
+//! ```
+//!
+//! ## Handedness
+//!
+//! **Handedness** (also called chirality) is this weird property of 3D things.
+//! If two 3D things have different handedness, then it is impossible to rotate
+//! or translate one to look like the other; it's only possible if you mirror
+//! it.
+//!
+//! Most often, in games and 3D modeling, this corresponds to which way the Z
+//! axis grows. Often, +X is right, +Y is up, and the handedness determines
+//! whether +Z is into or out of the screen.
+//!
+//! See wikipedia for more information on handedness:
+//! - <https://en.wikipedia.org/wiki/Right-hand_rule>
+//! - <https://en.wikipedia.org/wiki/Orientation_(vector_space)>
+//!
+//! ## Orientation
+//!
+//! The **orientation** of the coordinate system determines which way textures
+//! are displayed on cube faces. Most often, the orientation is "Y up", meaning
+//! that textures are displayed with the top of the image toward the +Y axis.
+//!
+//! # `{N, U, V}` Space
+//!
+//! "`{N, U, V}` space" is basically the **face-local coordinate space**
```

Comment:
The proper math jargon is "Euclidean tangent space".

Reply:
Oh, thanks! That connects some bridges in my brain :)